test/SlowTests/Server/Replication/ReplicationCleanTombstones.cs

```diff
@@ -879,5 +862,53 @@ public async Task CanBackupAndRestoreTombstonesWithSameId(Options options)
                 }
             }
         }
+
+        private async Task<long> CreateAndRunBackup(IDocumentStore store, RavenDatabaseMode mode, string backupPath)
+        {
+            var config = Backup.CreateBackupConfiguration(backupPath);
+
+            if (mode == RavenDatabaseMode.Single)
+                return await Backup.UpdateConfigAndRunBackupAsync(Server, config, store);
+            
+            var backupCompleted = await Sharding.Backup.WaitForBackupToComplete(store);
+            var backupTaskId = await Sharding.Backup.UpdateConfigurationAndRunBackupAsync(Server, store, config);
+            Assert.True(WaitHandle.WaitAll(backupCompleted, TimeSpan.FromSeconds(10)));
+
+            return backupTaskId;
+        }
+
+        private async Task<RestoreBackupConfiguration> RunBackupAndCreateRestoreConfiguration(DocumentStore store, RavenDatabaseMode mode, long backupTaskId, string backupPath)
+        {
+            var databaseName = GetDatabaseName();
+            var configuration = new RestoreBackupConfiguration { DatabaseName = databaseName };
+
+            if (mode == RavenDatabaseMode.Single)
+            {
+                await Backup.RunBackupAndReturnStatusAsync(Server, backupTaskId, store, isFullBackup: false);
+                configuration.BackupLocation = Directory.GetDirectories(backupPath).First();
+            }
+            else
+            {
+                var backupCompleted = await Sharding.Backup.WaitForBackupToComplete(store);
+                await Sharding.Backup.RunBackupAsync(store.Database, backupTaskId, isFullBackup: false, servers: new List<RavenServer> { Server });
+                Assert.True(WaitHandle.WaitAll(backupCompleted, TimeSpan.FromSeconds(10)));
+
+                var dirs = Directory.GetDirectories(backupPath);
+                var sharding = await Sharding.GetShardingConfigurationAsync(store);
+                var settings = Sharding.Backup.GenerateShardRestoreSettings(dirs, sharding);
+                configuration.ShardRestoreSettings = settings;
+            }
+
+            return configuration;
+        }
+
+        private async Task<DocumentsStorage> GetDocumentStorageFor(IDocumentStore store, RavenDatabaseMode mode, string databaseName, string id)
```

Comment:
can we reuse `GetDocumentDatabaseInstanceForAsync`?

Reply:
done